File: 02-backend/spring-boot-ecommerce/src/main/java/com/ShopNook/ecommerce/dto/Purchase.java
package com.ShopNook.ecommerce.dto;

import com.ShopNook.ecommerce.entity.*;
import lombok.Data;

import java.util.*;

@Data
public class Purchase
{
    private Customer customer;
    private Address deliveryAddress;
    private Address paymentAddress;
    private Order order;
    private Set<OrderItem> orderItemSet;

}

File: 03-frontend/angular-ecommerce/src/app/utilities/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase 
{
    customer: Customer;
    deliveryAddress: Address;
    paymentAddress: Address;
    order: Order;
    orderItemSet: OrderItem[]; 

    constructor() {
        this.customer = new Customer();
        this.deliveryAddress = new Address();
        this.paymentAddress = new Address();
        this.order = new Order();
        this.orderItemSet = [];
    }
}
